commit/src/domain.rs

```diff
@@ -190,11 +169,14 @@ impl<Val: TwoAdicField> PolynomialSpace for TwoAdicMultiplicativeCoset<Val> {
     /// Then we decompose `gH` into `gK, ghK, gh^2K, ..., gh^{num_chunks}K`.
     fn split_domains(&self, num_chunks: usize) -> Vec<Self> {
         let log_chunks = log2_strict_usize(num_chunks);
-        debug_assert!(log_chunks <= self.log_n);
+        debug_assert!(log_chunks <= self.log_size());
         (0..num_chunks)
-            .map(|i| Self {
-                log_n: self.log_n - log_chunks,
-                shift: self.shift * self.subgroup_generator().exp_u64(i as u64),
+            .map(|i| {
+                Self::new(
+                    self.shift() * self.subgroup_generator().exp_u64(i as u64),
+                    self.log_size() - log_chunks,
+                )
+                .unwrap()
```

Comment:
It's pretty obvious that that this `.unwrap()` is totally safe but I think it's good to add a very short comment saying why. (Should make it easier in future to determine which `.unwrap()`'s we should try to remove and which we can ignore).
```suggestion
                .unwrap() // This won't panic as `self.log_size() - log_chunks < self.log_size() < Val::TWO_ADICITY`
```